lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -1043,30 +1039,24 @@ export class UserAgentApplication {
         } else {
             // Redirect cases
             tokenResponseCallback = null;
+            window.location.hash = "";
             // if set to navigate to loginRequest page post login
             if (this.config.auth.navigateToLoginRequestUrl) {
```

Comment:
I think it might be easier to pull this check and handling code out of the handleAuthenticationResponse function and perform these checks in the constructor or a separate function.

Reply:
@pkanher617 I separated handleAuthenticationResponse into 2 functions. One for the iframe/popup use cases and the other for redirect use cases. I think the flow is easier to follow and understand. Thanks for the suggestion